src/ktopt_planning_context.cpp

```diff
@@ -175,37 +183,114 @@ void KTOptPlanningContext::setRobotDescription(std::string robot_description)
+  auto [plant, scene_graph] = AddMultibodyPlantSceneGraph(builder.get(), 0.0);
 
   // TODO:(kamiradi) Figure out object parsing
   // auto robot_instance = Parser(plant_, scene_graph_).AddModelsFromString(robot_description_, ".urdf");
 
   // HACK: For now loading directly from drake's package map
-  const char* ModelUrl =
-    "package://drake_models/franka_description/"
-    "urdf/panda_arm.urdf";
+  const char* ModelUrl = "package://drake_models/franka_description/"
+                         "urdf/panda_arm.urdf";
   const std::string urdf = PackageMap{}.ResolveUrl(ModelUrl);
-  auto robot_instance = Parser(plant_, scene_graph_).AddModels(urdf);
-  plant_->WeldFrames(plant_->world_frame(), plant_->GetFrameByName("panda_link0"));
+  auto robot_instance = Parser(&plant, &scene_graph).AddModels(urdf);
+  plant.WeldFrames(plant.world_frame(), plant.GetFrameByName("panda_link0"));
+
+  // planning scene transcription
+  const auto scene = getPlanningScene();
+  transcribePlanningScene(*scene);
+
 
   // for now finalize plant here
-  plant_->Finalize();
+  plant.Finalize();
+
+  // Apply MeshCat visualization
+  VisualizationConfig config;
+  ApplyVisualizationConfig(config, builder.get(), /*lcm_buses*/ nullptr, &plant, &scene_graph, meshcat_);
+
+  MeshcatVisualizerParams meshcat_viz_params;
+  auto& visualizer =
+      MeshcatVisualizer<double>::AddToBuilder(builder.get(), scene_graph, meshcat_,
+                                              std::move(meshcat_viz_params));
+  visualizer_ = &visualizer;
 
   // in the future you can add other LeafSystems here. For now building the
   // diagram
-  auto diagram_ = builder.Build();
+  diagram_ = builder->Build();
   diagram_context_ = diagram_->CreateDefaultContext();
-  plant_context_ = &diagram_->GetMutableSubsystemContext(
-    *plant_,
-    diagram_context_.get());
-  nominal_q_ = plant_->GetPositions(*plant_context_);
+  auto& plant_context = diagram_->GetMutableSubsystemContext(plant, diagram_context_.get());
+
+  nominal_q_ = plant.GetPositions(plant_context);
+
+  auto& vis_context = visualizer_->GetMyContextFromRoot(*diagram_context_);
+  visualizer_->ForcedPublish(vis_context);
+}
+
+void KTOptPlanningContext::transcribePlanningScene(const planning_scene::PlanningScene& planning_scene)
+{
+  RCLCPP_INFO(getLogger(), "trying something ...");
+  try
+  {
+    auto world = planning_scene.getWorld();
+  }
+  catch (const std::exception& e)
+  {
+    RCLCPP_ERROR_STREAM(getLogger(), "caought exception ... " << e.what());
+  }
+  auto& plant =
+    dynamic_cast<MultibodyPlant<double>&>(builder->GetMutableSubsystemByName("plant"));
+  auto& scene_graph =
+    dynamic_cast<SceneGraph<double>&>(builder->GetMutableSubsystemByName("scene_graph"));
+  for (const auto& object : planning_scene.getWorld()->getObjectIds())
+  {
+    const auto& collision_object = planning_scene.getWorld()->getObject(object);
+    if (!collision_object)
+    {
+      RCLCPP_ERROR(getLogger(), "No collision object");
```

Comment:
Maybe this is more of a warning or info? An empty planning scene (except the robot) could be a valid case